core/src/Streamly/Internal/Data/Stream/Reduce.hs

```diff
@@ -408,11 +408,11 @@ parseManyTill = undefined
 --
 {-# INLINE parseIterate #-}
 parseIterate
-    :: MonadThrow m
-    => (b -> Parser a m b)
-    -> b
+    :: MonadFail m
+    => (Either ParseError b -> Parser a m (Either ParseError b))
+    -> Either ParseError b
     -> Stream m a
-    -> Stream m b
+    -> Stream m (Either ParseError b)
```

Comment:
That's a terrible signature. See the change in parseMany signature. The signature should be:

```
parseIterate ::  (b -> Parser a m b) -> b -> Stream m a -> Stream m (Either ParseError b)
```